101-synapse-poc/azuredeploy.json

```diff
@@ -0,0 +1,621 @@
+            "metadata": {
+                "description": "'True' deploys an Apache Spark pool as well as a SQL pool. 'False' does not deploy an Apache Spark pool."
+            }
+        },
+        "sparkNodeSize": {
+            "type": "string",
+            "metadata": {
+                "description": "This parameter will determine the node size if SparkDeployment is true"
+            },
+            "defaultValue": "Medium",
+            "allowedValues": [
+                "Small",
+                "Medium",
+                "Large"
+            ]
+        },
+        "deploymentType": {
+            "type": "string",
+            "allowedValues": [
+                "devtest",
+                "poc",
+                "prod",
+                "shared"
+            ],
+            "defaultValue": "poc",
+            "metadata": {
+                "description": "Specify deployment type: DevTest, POC, Prod, Shared. This will also be used in the naming convention."
+            }
+        },
+        "sqlAdministratorLogin": {
+            "type": "string",
+            "defaultValue": "sqladminuser",
+            "metadata": {
+                "description": "The username of the SQL Administrator"
+            }
+        },
+        "sqlAdministratorLoginPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The password for the SQL Administrator"
+            }
+        },
+        "storageRoleUniqueId": {
+            "type": "string",
+            "defaultValue": "[newGuid()]",
+            "metadata": {
+                "description": "Leave this as default value. Generates a new GUID for role assignment."
+            }
+        },
+        "sku": {
+            "type": "string",
+            "defaultValue": "DW100c",
+            "allowedValues": [
+                "DW100c",
+                "DW200c",
+                "DW300c",
+                "DW400c",
+                "DW500c",
+                "DW1000c",
+                "DW1500c",
+                "DW2000c",
+                "DW2500c",
+                "DW3000c"
+            ],
+            "metadata": {
+                "description": "Select the SKU of the SQL pool."
+            }
+        },
+        "storageBlobDataContributorID": {
+            "type": "string",
+            "defaultValue": "ba92f5b4-2d11-453d-a403-e96b0029c9fe",
+            "allowedValues": [
+                "ba92f5b4-2d11-453d-a403-e96b0029c9fe"
+            ],
+            "metadata": {
+                "description": "Leave this as default value. Role ID for Storage Blob Data Contributor"
+            }
+        },
+        "metadataSync": {
+            "type": "bool",
+            "defaultValue": false,
+            "metadata": {
+                "description": "Choose whether you want to synchronise metadata."
+            }
+        },
+        "pauseGuid": {
+            "type": "string",
+            "defaultValue": "[newGuid()]",
```

Comment:
if this is for a roleAssignment (as it says in metadata.json) it must be idempotent and unique, and should be a variable (so the user can't break it)